Update spring-boot to 2.7.9

MichaelsJP opened the pull request:
<details>
<summary>Details</summary>



</details>

### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.

Things to check before merge:

- [x] check the base api url consistency (/ors/v2 vs /v2).
	- the `context-path` setting enforces /ors also in production when run with maven. This is a breaking change and should be avoided. The easier way is to remove the /ors completely. That means local dev setups won't have this path.
- [x] change `run_api_tests.sh` to curl for `v2/status` instead of `ors.run`
- [x]  check for correct config file configurations (run path of spring -> built-in tomcat is different from tomcat7 -> spring )
- [x] Add dev setup documentation for intelllij
- [x] Fix docker setup

Fixes #1151 
Closes #1026
closes #1285 
Partially touches #1025

### Information about the changes
- Key functionality added: Upgrade to spring-boot 2.7.9. Partially, the code wasn't properly using java11 and had to be fixed.
- Reason for change: The current spring-boot version is outdated and not maintained anymore.


### Required changes to ors config (if applicable)
- The new spring-boot update requires us to use the built-in tomcat runner. That runner has a different run path than the original tomcat7 servlet container. The testing config needs to be adjusted for this.

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

MichaelsJP commented:
Supersedes the old PR #1285 to spring 2.7.8

Closes #1285

jlink reviewed (approved):
LGTM as far as I'm competent which is only a little

MichaelsJP committed:
build(deps): upgrade spring-boot to 2.7.9

This upgrades spring-boot to the latest 2 version. Several parts had to be adapted to fit the new spring-boot API.

MichaelsJP committed:
test(api-tests): enhance api-tests

These changes let ors start the api-test routine with random ports and pass this port to the api-test execution. Therefore the api-tests can be executed non-blocking.

MichaelsJP committed:
chore(deps): update api-test dependencies

MichaelsJP committed:
docs(changelog): add info for spring-boot migration

MichaelsJP committed:
refactor(backend): remove illegal System.runFinalization call

For whatever reason System.runFinalization(); was called at context closure. This cuts short the native java closing routines, e.g. garbage collector etc. This is handled by spring-boot through the Bean context.

takb committed:
Merge branch 'master' into spring-update-to-2.7.9

takb committed:
test: fix path in ors-config-test.json

takb committed:
build(mvn): make spring-boot:run debuggable

add fork=false to spring-boot-maven-plugin to enable running in the IntelliJ debugger

takb committed:
Merge branch 'master' into spring-update-to-2.7.9

takb committed:
fix: correct ServletContextListener initialization

MichaelsJP committed:
build(spring-boot): ensure bean initialization order

The beans depend on each other and need to be initialized in order. Else the startup can randomly fail.

MichaelsJP committed:
fix(api): add /ors to the context-path

If the context-path is not defined with /ors it will fallback to /.

MichaelsJP committed:
test(debug): Temporal change to show the running path

jlink commented on openrouteservice/src/main/java/org/heigit/ors/api/converters/APIRequestProfileConverter.java:
```diff
@@ -15,17 +15,23 @@
 
-import org.springframework.core.convert.converter.Converter;
 import org.heigit.ors.api.requests.common.APIEnums;
+import org.jetbrains.annotations.NotNull;
```
Did that get in on purpose or by accident?

jlink commented on openrouteservice/src/main/resources/application.properties:
```diff
@@ -1,2 +1,6 @@
+# Workaround for failing springfox-swagger2.
+# springfox-swagger2 should be replaced by the maintained springdoc-openapi
+spring.mvc.pathmatch.matching-strategy = ANT_PATH_MATCHER
+server.servlet.context-path=/ors
```
Will fix context path for WAR file as well. Should be removed.

jlink reviewed (approved):
LGTM. Complex running of ITs can be handled in other PR.

MichaelsJP committed:
test(api-tests): run tests based on health status code

Before the api tests were waiting for 'ors.run' to be created at some place. Now it looks at the /v2/health api endpoints and checks for http 200. This indicates the graphs are build and the system is ready. Timeout is set to 4 minutes

MichaelsJP committed:
build(spring-boot): remove fork setting

The fork setting set to false mixes up run paths but allows for local debug session. When set to true it disables the ability for a local debug session but has the expected run paths. To have local debug sessions one can use  -Dspring-boot.run.fork=false to disable forking when needed. This is only possible when the fork settings is not set at all in the pom.

MichaelsJP committed:
chore(backend): remove annotation

intellij annotation was set by accident.

MichaelsJP committed:
docs(documentation): correct the java version to 11.

MichaelsJP committed:
build(spring-boot): remove `provided` from spring-boot-starter-actuator

spring-boot-starter-actuator is needed at runtime.

MichaelsJP committed:
test(api-tests): fix the api-test url path

MichaelsJP committed:
docs(documentation): Adjust the spring-boot:run examples

takb committed:
test(API): cleanup result test logging

MichaelsJP committed:
Merge branch 'master' into spring-update-to-2.7.9

takb committed:
fix(startup): exit gracefully on missing or invalid configuration file

MichaelsJP committed:
Merge branch 'master' into spring-update-to-2.7.9

takb reviewed (approved):
LGTM
